### PR TITLE
Added SwiftPM build of Siesta

### PR DESCRIPTION
**Pull request by pcantrell**
TL;DR: Attempting to add SwiftPM tests only for the newest version of Siesta, but preserve the existing Xcode-only config for older versions. Is this PR the proper way to accomplish that?

---

Siesta has been in the compatibility suite since the suite’s earliest days, as an Xcode-only project.

Siesta’s test suite depends on Quick and Nimble. When building as an Xcode project, Carthage manages those dependences, and swift-source-compat-suite thus can’t run the tests. However, at long last, I got Siesta working with SwiftPM (yay!), and now swift-source-compat-suite is capable of running Siesta’s tests via SwiftPM.

There is value in adding those tests to your suite, I think: there are some quirks in them you probably want in your regression tests. (In fact, as of last week, Siesta’s tests exposed a regression in the current dev snapshot.)

**BUT** only the very latest release of Siesta supports SwiftPM. I thus want to configure `BuildSwiftPackage` and `TestSwiftPackage` only for the latest releases, while still having the older ones run Xcode-only. My solution (in this PR) is to list Siesta twice with two different `path`s pointing at the same GitHub repo:

- one (already in the compat suite) configured to build only with Xcode, and pointing at both new & old versions of Siesta, and
- one (added in this PR) configured to build only with SwiftPM, and pointing only at the newer version of Siesta.

Does that make sense? Is there a better way?

### CODE REVIEW

**Comment by clackary** [comment]
Yes, I think it does make sense to have a separate entry in this case. Thinking ahead a bit, if there's ever a configuration that supports both Xcode/SwiftPM builds, it would be nice to combine them.

**Comment by clackary** [comment]
@swift-ci test

**Comment by pcantrell** [comment]
> Thinking ahead a bit, if there's ever a configuration that supports both Xcode/SwiftPM builds, it would be nice to combine them.

All Siesta release from here out support both. I just didn’t want to lose the 4.0 (and older 5.1 commit) from the suite. If you ever drop pre-5.1 from the suite, they could merge again.

**Comment by clackary** [comment]
Oh cool, in that case I'd suggest that we do something like...

path: siesta-legacy
 - 4.0
 - Xcode project actions

path: siesta
 - 5.1
 - Xcode project actions
 - swiftpm action

**Comment by clackary** [comment]
I think this setup would make it easier to add new configurations, as well as clean up older configs if/when they need to be cleaned up.

**Comment by pcantrell** [comment]
Fair enough! I’ll get that pushed as soon as I can find 5 min to test locally….

**Comment by clackary** [comment]
@swift-ci test

**Comment by pcantrell** [comment]
Done. LMK if that works.

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
Thanks @pcantrell!

**Comment by pcantrell** [comment]
Thanks likewise for maintaining this suite!
